tests/test_macros.py

```diff
@@ -0,0 +1,92 @@
+import dataclasses
+import os
+
+import dbt.tracking
+import pytest
+from _pytest.fixtures import SubRequest
+from dbt.adapters.factory import get_adapter, register_adapter, AdapterContainer
+from dbt.clients.jinja import MacroGenerator
+from dbt.config.runtime import RuntimeConfig
+from dbt.context import providers
+from dbt.contracts.connection import ConnectionState
+from dbt.contracts.graph.manifest import Manifest
+from dbt.adapters.spark.connections import (
+    SparkConnectionManager,
+    PyodbcConnectionWrapper,
+)
+from dbt.parser.manifest import ManifestLoader
+from dbt.tracking import User
+from pyspark.sql import SparkSession
+from sodaspark.scan import Connection
+
+
+dbt.tracking.active_user = User(os.getcwd())
+
+
+class _SparkConnectionManager(SparkConnectionManager):
+    @classmethod
+    def open(cls, connection):
+        handle = PyodbcConnectionWrapper(Connection())
+        connection.handle = handle
+        connection.state = ConnectionState.OPEN
+        return connection
+
+
+@dataclasses.dataclass(frozen=True)
+class Args:
+    project_dir: str = os.getcwd()
+
+
+@pytest.fixture
+def config() -> RuntimeConfig:
+    args = Args()
+
+    # Sets the Spark plugin in dbt.adapters.factory.FACTORY
+    config = RuntimeConfig.from_args(args)
```

Comment:
`dbt` is written as a command line tool. The entrypoint of dbt - the dbt cli - uses the parsed arguments heavily. I tried to find a balance between reusing what is in `dbt-core` with minimal dependencies: here we still use the `from_args` class method to create a `RuntimeConfig` - but with a smaller `Args` class.

Reply:
Heard. For our part, we know that:
- `RuntimeConfig` is a messy grab-bag
- We need to decouple the CLI / arg parsing logic from the actual entrypoints to the `dbt-core` application: https://github.com/dbt-labs/dbt-core/issues/4179 and https://github.com/dbt-labs/dbt-core/issues/4475